radix.py

```diff
@@ -0,0 +1,108 @@
+def radixsort(input_list):
+    """Sort list of integers and strings with radix sort algorithm."""
+    if type(input_list[0]) is str:
+        # for case when sorting string
+        base = 256
+        string = True
+        place = 0
+    else:
+        string = False
+        base = 10  # The number of buckets to use.
+        place = 1  # IE, the 1's, 10's, or 100's place.
+    hold = -1  # A placeholder
+
+    complete = False
+    while not complete:
+        complete = True
+
+        # Declare the number of buckets to use and create them.
+        buckets = [list() for _ in xrange(base)]
+
+        # Segment the input list into appropriate buckets.
+        for item in input_list:
+            if string:
+                if place < len(item):
+                    hold = ord(item[place])
+                else:
+                    hold = 0
+            else:
+                hold = item // place
+
+            # For each item in the list, append it into a bucket.
+            # To determine which bucket it should go into, divide the item
+            # by the base of the dataset in use.
+            # IE: 
+            # - an item of value 1 % 10 --> bucket 1
+            # - an item of value 5 % 10 --> bucket 5
+            buckets[int((hold % base))].append(item)
+
+            # Hold will be greater than 0 while there is a value in the list
+            # that has a number in the decimal place that has not already been
+            # iterated through.
+            if string:
+                if complete and hold > len(item):
+                    complete = False
+            else:
+                if complete and hold > 0:
+                    complete = False
```

Comment:
again, these two forks of logic are agonizingly similar.  What could we do to avoid needing the if/else?